### DISCUSSION TITLE
Error installing foundry

### DISCUSSION THREAD

**Discussion by HonourJAH** [Cyfrin/foundry-full-course-cu#2333] [Q&A]
ERROR:
curl: (23) Failure writing output to destination

FULL COMMAND
curl -L https://foundry.paradigm.xyz | bash
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   167  100   167    0     0    107      0  0:00:01  0:00:01 --:--:--   107
100  2189  100  2189    0     0    719      0  0:00:03  0:00:03 --:--:--  2017
Installing foundryup...
curl: (23) Failure writing output to destination

I got the error above while trying to install foundry
Kindly help

**Comment by EngrPips** [comment]
Hello @HonourJAH, check out this discussion https://askubuntu.com/questions/1387141/curl-23-failure-writing-output-to-destination maybe it will help.

**Reply by HonourJAH** [reply]
Thank you so much @EngrPips 
it worked!

**Reply by EngrPips** [reply]
Awesome, I am glad that helped you.

**Reply by HonourJAH** [reply]
can we connect please?

it seems you have really gone far with the course.

**Reply by HonourJAH** [reply]
You can drop your X handle for a follow up if you don't mind

**Reply by EngrPips** [reply]
I am @engrpips on Twitter. I am not always active on social media tho. I am mostly active on Cyfrin Github and Cyfrin Discord

**Reply by HonourJAH** [reply]
Alright then
Thank you

**Reply by EngrPips** [reply]
You are welcome, Friend.

**Comment by hassanelashiry** [comment]
**These are the steps that solved my installation problem**


1. **Try Using `wget` Instead:**

   If `curl` is not working, you can use `wget` to download the script:

   ```bash
   wget -O foundryup-install.sh https://foundry.paradigm.xyz
   ```

2. **Make the Script Executable:**

   After downloading, ensure the script is executable:

   ```bash
   chmod +x foundryup-install.sh
   ```

3. **Run the Script:**

   Execute the script to perform the installation:

   ```bash
   ./foundryup-install.sh
   ```

4. **Check the Script Contents:**

   Verify the script to ensure it contains the expected content. Open it with a text editor:

   ```bash
   nano foundryup-install.sh
   ```

   Confirm that the script content is what you intended to run.

5. **Update Your System:**

   Sometimes issues can arise from outdated packages. Update your system and `curl`:

   ```bash
   sudo apt update
   sudo apt upgrade
   sudo apt install curl
   ```

6. **The last step:**
   Run `foundryup` command. 
   ```bash
   foundryup
   ```

**Reply by HonourJAH** [reply]
Thank you so much, @hassanelashiry

**Reply by hassanelashiry** [reply]
you are welcome, bro!
